File: courses-project/back-end/src/main/java/br/com/courses/service/course/CourseService.java
package br.com.courses.service.course;

import br.com.courses.domain.course.CourseExibitionData;
import br.com.courses.domain.course.CourseRegisterDTO;
import br.com.courses.domain.course.CourseResponseData;
import br.com.courses.domain.subscribe.RegisterSubscriptionDTO;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class CourseService implements ICourseService {

    @Override
    public CourseResponseData create(CourseRegisterDTO courseRegisterDTO) {
        return null;
    }

    @Override
    public List<CourseExibitionData> getAll() {
        return null;
    }

    @Override
    public void subscribe(RegisterSubscriptionDTO registerSubscriptionDTO) {
    }


}

File: courses-project/front-end/src/services/courses/CourseService.ts
import Service from "../Service";
import {AxiosResponse} from "axios";
import Course from "../../models/courses/Course";
import RegisterSubscriptionDTO from "../../models/courses/RegisterSubscriptionDTO";


const CourseService =  {

    getAll: async (): Promise<AxiosResponse<Course[]>> => {

        let response: AxiosResponse = await Service.get('/courses');

        return response;

    },
    subscribe: async (subscriptionDTO: RegisterSubscriptionDTO):  Promise<AxiosResponse> => {

        const response: AxiosResponse =  await Service.post('/courses/subscribe', subscriptionDTO);

        return response;

    }

}


export default CourseService;
